### ISSUE TITLE
Loss function rewrite

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/nd4j#669]
Looking at the RMSE_XENT loss function as define here: https://github.com/deeplearning4j/nd4j/blob/master/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/lossfunctions/LossCalculation.java#L121

This appears to be incorrect. There is no cross entropy component, thus the ..._XENT name doesn't make sense. Furthermore, this appears to be an incorrect version of RMSE: https://en.wikipedia.org/wiki/Root-mean-square_deviation
If this was the correct RMSE, the sqrt should be after the sum, not before.

**Comment by eraly** [comment]
Changing owner on wrong issue. Oops. Back to me.

**Comment by eraly** [comment]
So like Alex said the RMSE_XENT loss function is wrong. I fixed it to be the correct definition of RMSE.
https://github.com/deeplearning4j/nd4j/compare/eraly

@AlexDBlack If changing this to the RMSE was what you had in mind I can merge and close.

**Comment by agibsonccc** [comment]
Yeah RMSE is fine.

**Comment by eraly** [comment]
Yeah. What do you guys think of taking this out in the next release? 
It might do more harm than good to leave it in. It's confusing and incorrect...
@AlexDBlack @agibsonccc @nyghtowl

**Comment by nyghtowl** [comment]
If its wrong, I'd go with deprecating for next release and then removal after. Is the plan to replace this with the correct form based on the loss function re-write or remove completely as an option?

**Comment by agibsonccc** [comment]
I'm not quite ready for a 0.6.0..we can remove it after we hit that
milestone. Keep it deprecated for the next few releases (basically anything
that's 0.5.x)

**Comment by agibsonccc** [comment]
Released with 0.6.0
